code/modules/mob/living/simple_animal/simple_animal.dm

```diff
@@ -366,3 +383,53 @@
 	return
 /mob/living/simple_animal/ExtinguishMob()
 	return
+
+
```

Comment:
the following functions make animal sleeping less buggy. there were lotsa bugs with it